packages/autorest.go/src/generator/models.ts

```diff
@@ -379,6 +380,9 @@ function generateJSONUnmarshallerBody(modelType: ModelType | PolymorphicType, re
   if (addlProps) {
     unmarshalBody += '\t\tdefault:\n';
     unmarshalBody += emitAddlProps('\t', addlProps);
+  } else if (options.disallowUnknownFields) {
+    unmarshalBody += '\t\tdefault:\n';
+    unmarshalBody += `\t\t\terr = fmt.Errorf("unmarshalling type %T, unknown field %q", ${receiver}, key)\n`;
```

Comment:
I'm +1 on this PR as is, but wanted to ask a question about this line to check we've thought it through enough.

At the moment our service doesn't return unmarshalling errors to the user verbatim, we just return a fairly general error.

If we enabled this toggle, to make full use of it, I think it'd imply starting passing through the error string directly from json.Unmarshal to the end user.  Is that what we want?
 
Leaking %T isn't wonderful, but being able to construct an error message saying "foo.properties[123].bar.badField is not allowed" would be a pain, so it seems pragmatic to me.

We wouldn't want a defined error type like UnrecognisedFieldError{ TypeName string, FieldName string } to be able to pick out this error?

Reply:
`"unmarshalling type %T, unknown field %q"` is a copy from the Go standard library JSON implementation.  This works for me, let's not over complicate.

Reply:
I thought about this, but opted to keep it simple and match what the json package does.
even the error string is the same in fact.